apps/protocol-frontend/src/hooks/useDaoUserUpdate.ts

```diff
@@ -0,0 +1,68 @@
+import { useUser } from '../contexts/UserContext';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useGovrnToast } from '@govrn/protocol-ui';
+
+interface DaoUserUpdateProps {
+  userId: number;
+  guildId: number;
+  favorite?: boolean;
+  membershipStatusId?: number;
+}
+
+export const useDaoUserUpdate = () => {
+  const toast = useGovrnToast();
+  const { govrnProtocol: govrn } = useUser();
+  const queryClient = useQueryClient();
+  const { mutateAsync, isLoading, isError, isSuccess } = useMutation(
+    async ({
+      userId,
+      guildId,
+      favorite,
+      membershipStatusId,
+    }: DaoUserUpdateProps) => {
+      const mutationData = await govrn.guild.user.update({
+        userId: userId,
+        guildId: guildId,
+        favorite: favorite,
+        membershipStatusId: membershipStatusId,
+      });
+      return { mutationData };
+    },
+    {
+      onSuccess: (_, { favorite }) => {
+        queryClient.invalidateQueries(['userDaos']);
+        queryClient.invalidateQueries(['daoUsersList']);
+
+        const toastSuccessId = 'dao-user-create-success';
+        if (!toast.isActive(toastSuccessId)) {
+          toast.success({
+            id: toastSuccessId,
+            title:
+              favorite !== undefined
+                ? 'Favorite Status Updated'
+                : 'DAO Updated',
+            description:
+              favorite !== undefined
+                ? 'Favorite status has been updated.'
+                : 'DAO has been updated.',
```

Comment:
I am not too much of a fan of this because it will not scale with additional fields to update